include/aspect/geometry_model/ellipsoidal_chunk.h

```diff
@@ -0,0 +1,254 @@
+        representative_point(const double depth) const;
+
+        virtual
+        double
+        maximal_depth() const;
+
+        /**
+         * Return the set of boundary indicators that are used by this model. This
+         * information is used to determine what boundary indicators can be used in
+         * the input file.
+         *
+         * The box model uses boundary indicators zero through 2*dim-1, with the first
+         * two being the faces perpendicular to the x-axis, the next two perpendicular
+         * to the y-axis, etc.
+         */
+        virtual std::set<types::boundary_id>
+        get_used_boundary_indicators() const;
+
+        /*
+        *Set symbolic names for boudaries (mrtf)
+        */
+        virtual std::map<std::string,types::boundary_id>
+        get_symbolic_boundary_names_map() const;
+
+        /**
+         * Declare the parameters this class takes through input files.
+         */
+        static
+        void
+        declare_parameters(ParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter
+         * file.
+         */
+        virtual
+        void
+        parse_parameters(ParameterHandler &prm);
+
+        /**
+         * calculate radius at current position
+         */
+        static
+        double
+        get_radius(const Point<dim> &point);
+
+        /**
+         * retrieve the semi minor axis b value
+         */
+        static
+        double
+        get_semi_minor_axis_b();
+
+        /**
+         * retrieve the semi major axis a value
+         */
+        static
+        double
+        get_semi_major_axis_a();
+
+
+        /**
+         * retrieve the value of the eccentricity
+         */
+        static
+        double
+        get_eccentricity();
+
+
+        /**
+         * retrieve the rot_para_to_para_angle variable
+         */
+        static
+        double
+        get_rot_para_to_para_angle();
+
+
+        /**
+         * retrieve the para_to_rect_angle variable
+         */
+        static
+        double
+        get_para_to_rect_angle();
+
+
+        /**
+         * retrieve the rotation_longitude variable
+         */
+        static
+        double
+        get_rotation_longitude();
+
+
+        /**
+         * retrieve the rotation_latitude variable
+         */
+        static
+        double
+        get_rotation_latitude();
+
+      private:
+        // Declare variables for reading in coordinates of the region of interest.
+        std::vector<std::vector<double> > corners;
+        std::vector<std::vector<double> > corners_parallelogram;
+        std::vector<std::vector<double> > corners_rectangle;
+        static double semi_major_axis_a;
+        static double Eccentricity;
+        static double semi_minor_axis_b;
+        static double rot_para_to_para_angle;
+        static double para_to_rect_angle;
+        static double rotation_longitude;
+        static double rotation_latitude;
```

Comment:
What is the reason that these variables are all `static`? Static variables are like global variables, and we know that global variables are generally frowned upon unless they are constants, but these here are not. Could you mark them as `const`? If not, please make them non-static and initialize them properly in the constructor.